pkg/controllers/node/node_controller.go

```diff
@@ -105,12 +117,16 @@ func NewNodeController(ctx context.Context, k8sClientset *kubernetes.Clientset,
 	// also syncs up labels between k8s/calico node objects
 	nc.indexer, nc.informer = cache.NewIndexerInformer(listWatcher, &v1.Node{}, 0, handlers, cache.Indexers{})
 
-	if syncLabels {
+	if nc.syncLabels {
 		// Start the syncer.
 		nc.initSyncer()
 		nc.syncer.Start()
 	}
 
+	if nc.autoHostEndpoints {
+		nc.syncAllHostendpoints()
```

Comment:
We need to make sure the syncer is started in this branch as well.

Right now, the syncer is only started if `syncLabels` is true.

Reply:
98eaaac